### ISSUE TITLE
Use python3 in the shebang line in acbfe.py

### ISSUE THREAD

**Issue by GeoRW** [ACBF-Advanced-Comic-Book-Format/ACBF-Editor#21]
Ubuntu (others) don't have python alias referring to python3, so we should use python3 in shebang. It is recommended per https://legacy.python.org/dev/peps/pep-0394/

**Comment by GeoRW** [comment]
So, even for linux distribution you want to create a bundled package with all dependencies including python interpreter?

**Comment by mizaki** [comment]
Yes, that way you don't have to worry about what people have installed on their PC (outside of the possible GTK4) problem. If ACBF-Editor doesn't work with python 3.12 and that's all they have installed, it doesn't matter etc. Making a Flatpak as well/instead might even be an idea.

**Comment by mizaki** [comment]
It's the build pipeline and you want to make a `.deb`? So, really, it's about setting up the build process. What is.was your build process like before? How is the actual python program packaged? There seems to be a clear `pyinstaller` to `deb` pathway [fpm](https://www.pythonguis.com/tutorials/packaging-pyqt5-applications-linux-pyinstaller/) is one example. And the ACBF Editor itself needs to be packaged?

If you only want to build debs and only use deb packages that is very limiting. Sure, making a new build process may be painful, but changing over to GTK4 already is :) There may be packages in PyPI that are very useful and instead of folding them in manually (or writing a script to do it), doesn't it make more sense to have the build action do it for us?

If you're using Ubuntu, you're using snaps whether you like it or not :) Flatpaks are becoming far more popular and instead of having to make various packages or various distros, it's one flatpak that's in all the "Software Centre"'s.

**Comment by GeoRW** [comment]
I want to target Ubuntu as default. If you have time to create and maintain build process for everything else, then fine.
There's no build process for GTK3 version, it just checks dependencies when it runs. That's the beauty of python under linux, you don't need a build to run it, it just runs out of the box. And it's easy for anyone to just run and try it out without creating python venv/cloning other git repository etc.
I created a simple install script for it in the past to install icons, desktop file etc., but I don't care if setuptools will be used instead or anything else.
For external libraries, it is good to check if it's really worth it and if not make it just optional dependency. With Kumiko, I didn't see a difference to simple algorithm which is used in GTK3 version. It may have better results when tested on various comics though, didn't tested it too much. I had another problem with it, that python failed to import stuff from kumiko subfolders for some reason. There's also https://github.com/Vrroom/segment-anything-comic repo which may have better results.

**Comment by GeoRW** [comment]
So, regarding dependencies:
1. If there's Ubuntu stable alternative use that one (that's the case of pycountry vs. isocodes
2. If there's no stable alternative make it optional dependency
3. If it's not possible to make it optional, implement directly into our code

In case of Kumiko, I would copy over their lib folder (maybe rename it to kumikolib) and their kumikolib.py. Then give all the credits in readme, licence file etc.

**Comment by mizaki** [comment]
1. That's find but I didn't want to be beholden to Ubuntu packages for what will be a multi-OS/distro program. I would still like to know the process you are using to make the `deb` package so it can be put into the build action (if required, I don't know how you get them onto Launchpad.)

There is a way in git to link and bring in other repo's code, so I'll look into that for kumiko (it worked better with some 2000AD pages I threw at it. Still not perfect but better.) Only thing is how that works with licenses. It's okay here are both are basically GNU v3.

**Comment by GeoRW** [comment]
Regarding deb, I will need to look into that, didn't do it for a while. But deb is just a package with some configuration, so nothing complex. I'm almost finished with GTK3 version testing and plan to release that before X-Mas.
Regarding Kumiko, don't link that. Rather add latest stable code to ACBF Editor directly. I don't see much activity there after last Dec release, but if they start changing things, with each commit they can introduce some bug. If you create a package and distribute some external library as part of it, the licence needs to allow it anyway.
If conservative approach is holding you back, then create experimental branch to experiment with some stuff there. If something works well, it can be added to main then. ACBF Editor is quite stable app, so I don't imagine too much other stuff is needed there for core functionality and if some feature brings many external dependencies I prefer to add it as optional dependency.
